### PR TITLE
🐛 Source Delighted: Fix pagination for `survey_responses`, `bounces` and `unsubscribes` streams

### LINKED ISSUE #9271: 🐛 Source Delighted: incorrect pagination handling in `survey_responses`, `bounces` and `unsubscribes` streams
Subtask for #8906.

## Current Behavior
`read` command only outputs 100 records for `survey_responses`, `bounces` and `unsubscribes` streams.

## Expected Behavior
All records should be read.

## Steps to Reproduce
1. Create 100+ records for one of `survey_responses`, `bounces` or `unsubscribes` streams;
2. Run `read` command for that stream;
3. See that only 100 records are read.

## Are you willing to submit a PR?
Yes.

### PR DESCRIPTION
**Pull request by Zirochkaa**
# Bug-fix Request

Fixes #9271.

---

## Reason
Right now pagination handling for `survey_responses`, `bounces` and `unsubscribes` streams works incorrectly.
Current implementation is the following: if we got `Link` response header then we extract `page_info` parameter from it and send it in the next request (like, `page_info=f54r54r4r`). If there is no `Link` response header then all data was read. This behavior works only for `people` stream.

## Confirmation
 - [x] **Was reproduced locally**

## How does the code change in the PR fix the issue?
For `survey_responses`, `bounces` and `unsubscribes` streams pagination handling should be following: send `page` parameter with number of page in it (like, `page=1`, `page=10` etc).

### CODE REVIEW

File: airbyte-integrations/connectors/source-delighted/source_delighted/schemas/survey_responses.json [hunk 1]
```diff
@@ -102,19 +116,5 @@
         "required": ["id", "value", "question"]
```
Comment (yevhenii-ldv) [inline airbyte-integrations/connectors/source-delighted/source_delighted/schemas/survey_responses.json:116]: Is this parameter needed here?

File: docs/integrations/sources/delighted.md [hunk 2]
```diff
@@ -37,4 +37,5 @@ This connector supports `API PASSWORD` as the authentication method.
 
 | Version | Date | Pull Request | Subject |
 | :--- | :--- | :--- | :--- |
+| 0.1.1   | 2022-01-04 | [9275](https://github.com/airbytehq/airbyte/pull/9275) | Fix pagination handling for `survey_responses`, `bounces` and `unsubscribes` streams |
```
Comment (yevhenii-ldv) [inline docs/integrations/sources/delighted.md:40]:
```suggestion
| 0.1.1 | 2022-01-04 | [9275](https://github.com/airbytehq/airbyte/pull/9275) | Fix pagination handling for `survey_responses`, `bounces` and `unsubscribes` streams |
```

File: airbyte-integrations/connectors/source-delighted/source_delighted/source.py [hunk 3]
```diff
@@ -77,38 +75,38 @@ def request_params(self, stream_state=None, **kwargs):
...
+    def path(self, **kwargs) -> str:
         return "people.json"
 
+    def next_page_token(self, response: requests.Response) -> Optional[Mapping[str, Any]]:
+        # Getting next page link
+        next_page = response.links.get("next", None)
+        if next_page:
+            return {"page_info": dict(parse_qsl(urlparse(next_page.get("url")).query)).get("page_info")}
+        else:
+            return None
```
Comment (yevhenii-ldv) [inline airbyte-integrations/connectors/source-delighted/source_delighted/source.py:87]:
```suggestion
```

File: airbyte-integrations/connectors/source-delighted/source_delighted/source.py [hunk 4]
```diff
@@ -32,12 +33,10 @@ def __init__(self, since: int, **kwargs):
...
 
     def next_page_token(self, response: requests.Response) -> Optional[Mapping[str, Any]]:
-        # Getting next page link
-        next_page = response.links.get("next", None)
-        if next_page:
-            return dict(parse_qsl(urlparse(next_page.get("url")).query))
-        else:
-            return None
+        response_data = response.json()
+        if response_data and len(response_data) == self.limit:
```
Comment (yevhenii-ldv) [inline airbyte-integrations/connectors/source-delighted/source_delighted/source.py:37]: Can `response_data ` be not an array?

**Review by yevhenii-ldv — CHANGES_REQUESTED** [review]
A little changes

File: airbyte-integrations/connectors/source-delighted/source_delighted/schemas/survey_responses.json (hunk 1 above)
Comment (Zirochkaa) [reply airbyte-integrations/connectors/source-delighted/source_delighted/schemas/survey_responses.json:116]: Nope, removed.

File: airbyte-integrations/connectors/source-delighted/source_delighted/source.py (hunk 4 above)
Comment (Zirochkaa) [reply airbyte-integrations/connectors/source-delighted/source_delighted/source.py:37]: No, it'll always an array.

File: airbyte-integrations/connectors/source-delighted/source_delighted/source.py (hunk 3 above)
Comment (Zirochkaa) [reply airbyte-integrations/connectors/source-delighted/source_delighted/source.py:87]: Done.

File: docs/integrations/sources/delighted.md (hunk 2 above)
Comment (Zirochkaa) [reply docs/integrations/sources/delighted.md]: Done.

**Review by yevhenii-ldv — APPROVED** [review]

**Pull request merged**